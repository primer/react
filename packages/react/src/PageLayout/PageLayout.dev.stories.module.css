.PageLayout {
  border: var(--borderWidth-default) solid red;
  color: var(--fgColor-success);
}
