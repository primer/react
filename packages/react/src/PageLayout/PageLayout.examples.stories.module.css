.ResponsiveFlash {
  display: none;
}

@media screen and (min-width: 768px) {
  .ResponsiveFlash {
    display: block;
  }
}

.BreadcrumbHeaderRow {
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: var(--base-size-16, 16px);
}

.HeaderRow {
  display: flex;
  justify-content: space-between;
}

.ResponsiveBox {
  display: none;
}

@media (min-width: 768px) {
  .ResponsiveBox {
    display: block;
  }
}
