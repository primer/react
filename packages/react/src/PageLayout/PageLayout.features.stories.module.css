.HeaderStack {
  display: flex;
  flex-direction: column;
  gap: var(--base-size-16);
}

.TitleHeading {
  font-weight: var(--base-text-weight-normal);
}

.TitleSubdued {
  color: var(--fgColor-muted);
  font-weight: var(--base-text-weight-light);
}

.StatusRow {
  display: flex;
  gap: var(--base-size-8);
  align-items: center;
}

.StatusMeta {
  font-size: var(--text-body-size-medium);
  color: var(--fgColor-muted);
}

.BoldMetaLink {
  font-weight: var(--base-text-weight-semibold);
}

.ContentBox {
  border: var(--borderWidth-default) solid var(--borderColor-default);
  border-radius: var(--borderRadius-medium);
  height: 200px;
}

.ScrollBox {
  max-width: 100%;
  overflow-x: auto;
  border: var(--borderWidth-default) solid var(--borderColor-default);
  white-space: nowrap;
  margin-top: var(--base-size-16);
  padding: var(--base-size-16);
  border-radius: var(--borderRadius-medium);
}

.PaneStack {
  display: flex;
  flex-direction: column;
  gap: var(--base-size-16);
}

.PaneSectionHeading {
  font-size: var(--text-body-size-small);
  font-weight: var(--base-text-weight-semibold);
  display: block;
  color: var(--fgColor-muted);
}

.PaneMetaText {
  font-size: var(--text-body-size-small, 12px);
  color: var(--fgColor-muted);
  /* stylelint-disable-next-line primer/typography */
  line-height: 1.25;
}

.PaneSeparator {
  width: 100%;
  height: 1px;
  /* stylelint-disable-next-line primer/colors */
  background-color: var(--borderColor-default);
}

.ContentGrid {
  display: grid;
  gap: var(--base-size-16);
}

.Paragraph {
  margin: 0;
}

.NestedScrollContainer {
  display: grid;
  grid-template-rows: auto 1fr auto;
  height: 100vh;
}

.OverflowAuto {
  overflow: auto;
}

.StickyHeader {
  position: sticky;
  top: 0;
  display: grid;
  place-items: center;
  background-color: var(--bgColor-muted);
  border-bottom: var(--borderWidth-default) solid var(--borderColor-default);
  z-index: 100;
}

.PaneHeading {
  font-size: var(--text-title-size-medium);
}
