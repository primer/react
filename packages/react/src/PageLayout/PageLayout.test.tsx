import {describe, it, expect, vi} from 'vitest'
import {page} from '@vitest/browser/context'
import {act, fireEvent, render, screen} from '@testing-library/react'
import 'react-intersection-observer/test-utils'
import {viewportRanges} from '../hooks/useResponsiveValue'
import {PageLayout} from './PageLayout'
import {Placeholder} from '../Placeholder'

describe('PageLayout', async () => {
  await page.viewport(1280, 800)
  it('renders default layout', () => {
    const {container} = render(
      <PageLayout>
        <PageLayout.Header>Header</PageLayout.Header>
        <PageLayout.Content>Content</PageLayout.Content>
        <PageLayout.Pane>Pane</PageLayout.Pane>
        <PageLayout.Footer>Footer</PageLayout.Footer>
      </PageLayout>,
    )
    expect(container).toMatchSnapshot()
  })

  it('renders condensed layout', () => {
    const {container} = render(
      <PageLayout padding="condensed" rowGap="condensed" columnGap="condensed">
        <PageLayout.Header>Header</PageLayout.Header>
        <PageLayout.Content>Content</PageLayout.Content>
        <PageLayout.Pane>Pane</PageLayout.Pane>
        <PageLayout.Footer>Footer</PageLayout.Footer>
      </PageLayout>,
    )
    expect(container).toMatchSnapshot()
  })

  it('renders with dividers', () => {
    const {container} = render(
      <PageLayout>
        <PageLayout.Header divider="line" dividerWhenNarrow="filled">
          Header
        </PageLayout.Header>
        <PageLayout.Content>Content</PageLayout.Content>
        <PageLayout.Pane position="start" divider="line" dividerWhenNarrow="filled">
          Pane
        </PageLayout.Pane>
        <PageLayout.Footer dividerWhenNarrow="line">Footer</PageLayout.Footer>
      </PageLayout>,
    )
    expect(container).toMatchSnapshot()
  })

  it('renders pane in different position when narrow', () => {
    const {container} = render(
      <PageLayout>
        <PageLayout.Header>Header</PageLayout.Header>
        <PageLayout.Content>Content</PageLayout.Content>
        <PageLayout.Pane positionWhenNarrow="start">Pane</PageLayout.Pane>
        <PageLayout.Footer>Footer</PageLayout.Footer>
      </PageLayout>,
    )
    expect(container).toMatchSnapshot()
  })

  // The test suite can't compute styles so skipping
  it.skip('can hide pane when narrow', () => {
    // Set narrow viewport
    act(() => {
      window.matchMedia(viewportRanges.narrow)
    })

    const {getByText} = render(
      <PageLayout>
        <PageLayout.Header>Header</PageLayout.Header>
        <PageLayout.Content>Content</PageLayout.Content>
        <PageLayout.Pane hidden={{narrow: true}}>Pane</PageLayout.Pane>
        <PageLayout.Footer>Footer</PageLayout.Footer>
      </PageLayout>,
    )

    expect(getByText('Pane')).not.toBeVisible()
  })

  // The test suite can't compute styles so skipping
  it.skip('shows all subcomponents by default', () => {
    // Set regular viewport
    act(() => {
      matchMedia(viewportRanges.regular)
    })

    const {getByText} = render(
      <PageLayout>
        <PageLayout.Header>Header</PageLayout.Header>
        <PageLayout.Content>Content</PageLayout.Content>
        <PageLayout.Pane hidden={{narrow: true}}>Pane</PageLayout.Pane>
        <PageLayout.Footer>Footer</PageLayout.Footer>
      </PageLayout>,
    )

    expect(getByText('Pane')).toBeVisible()
  })

  it('should support labeling landmarks through `aria-label`', () => {
    render(
      <PageLayout>
        <PageLayout.Header aria-label="Header">Header</PageLayout.Header>
        <PageLayout.Content aria-label="Content">Content</PageLayout.Content>
        <PageLayout.Pane>Pane</PageLayout.Pane>
        <PageLayout.Footer aria-label="Footer">Footer</PageLayout.Footer>
      </PageLayout>,
    )

    expect(screen.getByRole('banner')).toHaveAccessibleName('Header')
    expect(screen.getByRole('main')).toHaveAccessibleName('Content')
    expect(screen.getByRole('contentinfo')).toHaveAccessibleName('Footer')
  })

  it('should support labeling landmarks through `aria-labelledby`', () => {
    render(
      <PageLayout>
        <PageLayout.Header aria-labelledby="header-label">
          <span id="header-label">header</span>
        </PageLayout.Header>
        <PageLayout.Content aria-labelledby="content-label">
          <span id="content-label">content</span>
        </PageLayout.Content>
        <PageLayout.Pane>Pane</PageLayout.Pane>
        <PageLayout.Footer aria-labelledby="footer-label">
          <span id="footer-label">footer</span>
        </PageLayout.Footer>
      </PageLayout>,
    )

    expect(screen.getByRole('banner')).toHaveAccessibleName('header')
    expect(screen.getByRole('main')).toHaveAccessibleName('content')
    expect(screen.getByRole('contentinfo')).toHaveAccessibleName('footer')
  })

  describe('PageLayout.Pane', () => {
    it('should support a ref on the element wrapping the contents of Pane', () => {
      const ref = vi.fn()
      render(
        <PageLayout>
          <PageLayout.Pane ref={ref}>
            <div data-testid="content">Pane</div>
          </PageLayout.Pane>
        </PageLayout>,
      )
      expect(ref).toHaveBeenCalledWith(screen.getByTestId('content').parentNode)
    })

    it('should be resizable if `resizable` is set correctly', async () => {
      render(
        <PageLayout>
          <PageLayout.Pane resizable>
            <Placeholder height={320} label="Pane" />
          </PageLayout.Pane>
          <PageLayout.Content>
            <Placeholder height={640} label="Content" />
          </PageLayout.Content>
        </PageLayout>,
      )

      const placeholder = await screen.findByText('Pane')
      const pane = placeholder.parentNode
      const initialWidth = (pane as HTMLElement).style.getPropertyValue('--pane-width')
      const divider = await screen.findByRole('slider')

      // Moving divider should resize pane.
      fireEvent.focus(divider)
      //move it right 3 times
      fireEvent.keyDown(divider, {key: 'ArrowRight'})
      fireEvent.keyDown(divider, {key: 'ArrowRight'})
      fireEvent.keyDown(divider, {key: 'ArrowRight'})

      const finalWidth = (pane as HTMLElement).style.getPropertyValue('--pane-width')
      expect(finalWidth).not.toEqual(initialWidth)
    })
  })

  describe('PageLayout.Content', () => {
    it('should support a custom element type with the `as` prop', () => {
      const {container} = render(
        <PageLayout.Content as="div">
          <main>Content</main>
        </PageLayout.Content>,
      )
      expect(container.firstChild?.nodeName).toEqual('DIV')
    })
  })
})
