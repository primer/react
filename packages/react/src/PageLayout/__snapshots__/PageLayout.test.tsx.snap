// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout renders condensed layout 1`] = `
.c0 {
  padding: 16px;
}

.c1 {
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  width: 100%;
  margin-bottom: 16px;
}

.c3 {
  padding: 0;
}

.c4 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-top: 16px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  min-width: 1px;
}

.c7 {
  width: 100%;
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  width: 100%;
  margin-left: 0;
  margin-right: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 16px;
}

.c9 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-bottom: 16px;
}

.c10 {
  --pane-min-width: 256px;
  --pane-max-width-diff: 511px;
  --pane-max-width: calc(100vw - var(--pane-max-width-diff));
  width: 100%;
  padding: 0;
}

.c11 {
  height: 100%;
  position: relative;
  display: none;
  margin-right: 16px;
}

.c12 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  width: 100%;
  margin-top: 16px;
}

@media screen and (min-width:768px) {
  .c4 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:768px) {
  .c8 {
    width: auto;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    margin-left: 16px;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c10 {
    width: 256px;
    overflow: auto;
  }
}

@media screen and (min-width:1012px) {
  .c10 {
    width: 296px;
  }
}

@media screen and (min-width:1280px) {
  .c10 {
    --pane-max-width-diff: 959px;
  }
}

<div>
  <div
    class="c0"
    style="--sticky-pane-height: 100vh;"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        <div
          class="c3"
        >
          Header
        </div>
        <div
          class="c4"
        />
      </header>
      <div
        class="c5"
      >
        <main
          class="c6"
        >
          <div
            class=""
          />
          <div
            class="c7"
          >
            Content
          </div>
          <div
            class=""
          />
        </main>
        <div
          class="c8"
        >
          <div
            class="c9"
          />
          <div
            class="c10"
            style="--pane-width: 296px;"
          >
            Pane
          </div>
          <div
            class="c11"
          />
        </div>
      </div>
      <footer
        class="c12"
      >
        <div
          class="c9"
        />
        <div
          class="c3"
        >
          Footer
        </div>
      </footer>
    </div>
  </div>
</div>
`;

exports[`PageLayout renders default layout 1`] = `
.c0 {
  padding: 16px;
}

.c1 {
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  width: 100%;
  margin-bottom: 16px;
}

.c3 {
  padding: 0;
}

.c4 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-top: 16px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  min-width: 1px;
}

.c7 {
  width: 100%;
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  width: 100%;
  margin-left: 0;
  margin-right: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 16px;
}

.c9 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-bottom: 16px;
}

.c10 {
  --pane-min-width: 256px;
  --pane-max-width-diff: 511px;
  --pane-max-width: calc(100vw - var(--pane-max-width-diff));
  width: 100%;
  padding: 0;
}

.c11 {
  height: 100%;
  position: relative;
  display: none;
  margin-right: 16px;
}

.c12 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  width: 100%;
  margin-top: 16px;
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c0 {
    padding: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c4 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:1012px) {
  .c4 {
    margin-left: -24px;
    margin-right: -24px;
    margin-top: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c8 {
    width: auto;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    margin-left: 16px;
  }
}

@media screen and (min-width:1012px) {
  .c8 {
    margin-top: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c9 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:1012px) {
  .c9 {
    margin-left: -24px;
    margin-right: -24px;
    margin-bottom: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c10 {
    width: 256px;
    overflow: auto;
  }
}

@media screen and (min-width:1012px) {
  .c10 {
    width: 296px;
  }
}

@media screen and (min-width:1280px) {
  .c10 {
    --pane-max-width-diff: 959px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c11 {
    margin-right: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c12 {
    margin-top: 24px;
  }
}

<div>
  <div
    class="c0"
    style="--sticky-pane-height: 100vh;"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        <div
          class="c3"
        >
          Header
        </div>
        <div
          class="c4"
        />
      </header>
      <div
        class="c5"
      >
        <main
          class="c6"
        >
          <div
            class=""
          />
          <div
            class="c7"
          >
            Content
          </div>
          <div
            class=""
          />
        </main>
        <div
          class="c8"
        >
          <div
            class="c9"
          />
          <div
            class="c10"
            style="--pane-width: 296px;"
          >
            Pane
          </div>
          <div
            class="c11"
          />
        </div>
      </div>
      <footer
        class="c12"
      >
        <div
          class="c9"
        />
        <div
          class="c3"
        >
          Footer
        </div>
      </footer>
    </div>
  </div>
</div>
`;

exports[`PageLayout renders pane in different position when narrow 1`] = `
.c0 {
  padding: 16px;
}

.c1 {
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  width: 100%;
  margin-bottom: 16px;
}

.c3 {
  padding: 0;
}

.c4 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-top: 16px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  min-width: 1px;
}

.c7 {
  width: 100%;
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  width: 100%;
  margin-left: 0;
  margin-right: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 16px;
}

.c9 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-bottom: 16px;
}

.c10 {
  --pane-min-width: 256px;
  --pane-max-width-diff: 511px;
  --pane-max-width: calc(100vw - var(--pane-max-width-diff));
  width: 100%;
  padding: 0;
}

.c11 {
  height: 100%;
  position: relative;
  display: none;
  margin-right: 16px;
}

.c12 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  width: 100%;
  margin-top: 16px;
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c0 {
    padding: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c4 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:1012px) {
  .c4 {
    margin-left: -24px;
    margin-right: -24px;
    margin-top: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c8 {
    width: auto;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    margin-left: 16px;
  }
}

@media screen and (min-width:1012px) {
  .c8 {
    margin-top: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c9 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:1012px) {
  .c9 {
    margin-left: -24px;
    margin-right: -24px;
    margin-bottom: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c10 {
    width: 256px;
    overflow: auto;
  }
}

@media screen and (min-width:1012px) {
  .c10 {
    width: 296px;
  }
}

@media screen and (min-width:1280px) {
  .c10 {
    --pane-max-width-diff: 959px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c11 {
    margin-right: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c12 {
    margin-top: 24px;
  }
}

<div>
  <div
    class="c0"
    style="--sticky-pane-height: 100vh;"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        <div
          class="c3"
        >
          Header
        </div>
        <div
          class="c4"
        />
      </header>
      <div
        class="c5"
      >
        <main
          class="c6"
        >
          <div
            class=""
          />
          <div
            class="c7"
          >
            Content
          </div>
          <div
            class=""
          />
        </main>
        <div
          class="c8"
        >
          <div
            class="c9"
          />
          <div
            class="c10"
            style="--pane-width: 296px;"
          >
            Pane
          </div>
          <div
            class="c11"
          />
        </div>
      </div>
      <footer
        class="c12"
      >
        <div
          class="c9"
        />
        <div
          class="c3"
        >
          Footer
        </div>
      </footer>
    </div>
  </div>
</div>
`;

exports[`PageLayout renders with dividers 1`] = `
.c0 {
  padding: 16px;
}

.c1 {
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  width: 100%;
  margin-bottom: 16px;
}

.c3 {
  padding: 0;
}

.c4 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-top: 16px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  min-width: 1px;
}

.c7 {
  width: 100%;
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  width: 100%;
  margin-left: 0;
  margin-right: 0;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  margin-bottom: 16px;
}

.c9 {
  --pane-min-width: 256px;
  --pane-max-width-diff: 511px;
  --pane-max-width: calc(100vw - var(--pane-max-width-diff));
  width: 100%;
  padding: 0;
}

.c10 {
  height: 100%;
  position: relative;
  display: none;
  margin-left: 16px;
}

.c11 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  width: 100%;
  margin-top: 16px;
}

.c12 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-bottom: 16px;
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c0 {
    padding: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c4 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:1012px) {
  .c4 {
    margin-left: -24px;
    margin-right: -24px;
    margin-top: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c8 {
    width: auto;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-right: 16px;
  }
}

@media screen and (min-width:1012px) {
  .c8 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c9 {
    width: 256px;
    overflow: auto;
  }
}

@media screen and (min-width:1012px) {
  .c9 {
    width: 296px;
  }
}

@media screen and (min-width:1280px) {
  .c9 {
    --pane-max-width-diff: 959px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c10 {
    margin-left: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {
  .c11 {
    margin-top: 24px;
  }
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {
  .c12 {
    margin-left: 0 !important;
    margin-right: 0 !important;
  }
}

@media screen and (min-width:1012px) {
  .c12 {
    margin-left: -24px;
    margin-right: -24px;
    margin-bottom: 24px;
  }
}

<div>
  <div
    class="c0"
    style="--sticky-pane-height: 100vh;"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        <div
          class="c3"
        >
          Header
        </div>
        <div
          class="c4"
        />
      </header>
      <div
        class="c5"
      >
        <main
          class="c6"
        >
          <div
            class=""
          />
          <div
            class="c7"
          >
            Content
          </div>
          <div
            class=""
          />
        </main>
        <div
          class="c8"
        >
          <div
            class="c4"
          />
          <div
            class="c9"
            style="--pane-width: 296px;"
          >
            Pane
          </div>
          <div
            class="c10"
          />
        </div>
      </div>
      <footer
        class="c11"
      >
        <div
          class="c12"
        />
        <div
          class="c3"
        >
          Footer
        </div>
      </footer>
    </div>
  </div>
</div>
`;
